rust/core-lib/src/plugins/rpc_types.rs

```diff
@@ -127,10 +127,35 @@ pub struct Span {
 }
 
 #[derive(Serialize, Deserialize, Debug)]
+pub struct ScopeSpan {
+    pub start: usize,
+    pub end: usize,
```

Comment:
Yea, syntax highlighting is definitely a special case: in most others we wouldn't expect spans to be contiguous. If we run into performance issues, we can add a new protocol method to handle this case.